File: app/src/main/java/com/meoyawn/remotelove/api/model/RecentTracks.java
package com.meoyawn.remotelove.api.model;

import android.os.Parcel;
import android.os.Parcelable;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

/**
 * Created by adelnizamutdinov on 10/6/14
 */
@NoArgsConstructor
public class RecentTracks implements Parcelable {
  public @NotNull Track[] track;

  @Override public int describeContents() { return 0; }

  @Override public void writeToParcel(@NotNull Parcel dest, int flags) {
    dest.writeParcelableArray(this.track, flags);
  }

  private RecentTracks(Parcel in) {
    this.track = (Track[]) in.readParcelableArray(Track[].class.getClassLoader());
  }

  public static final Parcelable.Creator<RecentTracks> CREATOR = new Parcelable
      .Creator<RecentTracks>() {
    public RecentTracks createFromParcel(@NotNull Parcel source) {return new RecentTracks(source);}

    @NotNull public RecentTracks[] newArray(int size) {return new RecentTracks[size];}
  };
}

File: app/src/main/kotlin/com/meoyawn/remotelove/api/model/RecentTracks.kt
package com.meoyawn.remotelove.api.model

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Created by adelnizamutdinov on 10/3/14
 */
data class Image {
  JsonProperty("#text") var url: String = ""
  var size: String = ""
}

data class Artist {
  var name: String = ""
  var image: Array<Image> = array()
}

data class Attr {
  var nowplaying: Boolean = false
}

data class Track {
  var artist: Artist = Artist()
  var loved: Int = 0
  var name: String = ""
  var url: String = ""
  var image: Array<Image> = array()
  JsonProperty("@attr") var attr: Attr? = null

  class object {
    val EMPTY = Track()
  }
}

data class RecentTracks {
  var track: Array<Track> = array()
}

data class RecentTracksHolder {
  var recenttracks: RecentTracks = RecentTracks()
}
